src/backend/gporca/libgpopt/src/operators/CExpressionPreprocessor.cpp

```diff
@@ -84,7 +84,8 @@ CExpressionPreprocessor::PexprEliminateSelfComparison(CMemoryPool *mp,
 		return CPredicateUtils::PexprEliminateSelfComparison(mp, pexpr,
 															 pcrsNotNull);
 	}
-	if (pop->FLogical())
```

Comment:
squash the commit ffb8b2e1e8141c43d38a0eac37b879af7b7c415b to 4d2c7630bf5e06a783bc5b0ae67c9810c601784b pls. 

Current PR won't be squash when merged.